src/elm/Main.elm

```diff
@@ -1141,12 +1141,12 @@ viewContent model =
     case model.page of
         Pages.Overview ->
             ( "Overview"
-            , Pages.Home.view model.user ToggleFavorite
+            , lazy2 Pages.Home.view model.user homeMsgs
             )
 
         Pages.AddRepositories ->
             ( "Add Repositories"
-            , Pages.AddRepos.view model addReposMsgs
+            , lazy2 Pages.AddRepos.view model addReposMsgs
```

Comment:
for example, since we pass in a PartialModel to the AddRepos page, we will only trigger a DOM rerender when the `sourceRepos` field of the main model changes, even if say, the `theme` is updated

Reply:
we _are_ passing the whole model along here, so i don't think this would gain much as _lazy_ would evaluate the whole thing.. when we're passing fields of a model as we do above with `model.user`, i think we're getting the expected benefit. `AddRepos` accepts any record that at least has `user`, `sourceRepos`, and `filters` fields (see https://github.com/go-vela/ui/blob/master/src/elm/Pages/AddRepos.elm#L71-L73) and of course our model as a whole fits that criteria.

Reply:
just pushed some changes!